src/terraform-aws/content.md

```diff
@@ -0,0 +1,713 @@
+
+Managed Kubernetes services create and operate a cluster for you and give you access to it.
+
+_They allow you to use Kubernetes in a Software as a Service (SaaS) manner._
+
+Naturally, managed Kubernetes services provide the highest degree of automation (creation and operation are entirely done for you), but the least amount of flexibility (you can configure only those settings that the service provider exposes through its API).
+
+The most popular managed Kubernetes services are provided by the major cloud providers:
+
+- [Google Kubernetes Engine (GKE)](https://cloud.google.com/kubernetes-engine) by GCP
+- [Amazon Elastic Kubernetes Service (EKS)](https://aws.amazon.com/eks/) by AWS
+- [Azure Kubernetes Service (AKS)](https://docs.microsoft.com/azure/aks/) by Azure
+
+## Kubernetes installation tools
+
+Kubernetes installation tools allow you to install Kubernetes on your own infrastructure (both on-premises or in the cloud).
+
+_They allow you to use Kubernetes as a self-managed piece of software._
+
+Installation tools provide varying degrees of automation and flexibility depending on the extent to which the tool wants to "get it right for you".
+
+Currently, the most popular (and officially supported) Kubernetes installation tools are:
+
+- [kops](https://kops.sigs.k8s.io/)
+- [kubespray](https://kubespray.io/)
+- [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/)
+
+Let's briefly look at each of them.
+
+### kops
+
+Of the three presented tools, [kops](https://kops.sigs.k8s.io/) provides the highest degree of automation, but it's also the most opinionated one.
+
+What's special about kops is that it does not only install Kubernetes, but it also provisions the cloud infrastructure on which the cluster will run (currently, AWS and GCP are supported).
+
+That means, you can go from zero to a running cluster with effectively a single command.
+
+On the other hand, kops anticipates many decisions for you, thus reducing your flexibility for how you want your cluster to look like.
+
+For example, kops requires the name of the cluster to be a valid DNS name, and requires you to set up DNS records that resolve this name; you also have to create an [Amazon S3 bucket](https://aws.amazon.com/s3/) for kops to store its state.
+
+### kubespray
+
+[kubespray](https://kubespray.io/) is a collection of [Ansible](https://www.ansible.com/) playbooks that automate the installation of Kubernetes on existing infrastructure.
+
+That means, in contrast to kops, you need to already have a suitable infrastructure (either on-premises or in the cloud) before you can use kubespray.
+
+Once you have your infrastructure, kubespray provides a large number of options for configuring the Kubernetes installation.
+
+However, at the same time, kubespray attempts to create a "production-ready" cluster, which causes it to apply certain default settings and features (such as installing a CNI plugin by default).
```

Comment:
Not all options are configurable, though.

Kubespray attempts to create a "production-ready" cluster, which [...]